packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -376,6 +376,10 @@ class Overlay extends StatefulWidget {
 class OverlayState extends State<Overlay> with TickerProviderStateMixin {
   final List<OverlayEntry> _entries = <OverlayEntry>[];
 
+  /// Retrieve current list of [OverlayEntry]s.
+  @visibleForTesting
+  List<OverlayEntry> get entries => _entries;
```

Comment:
Some way of verifying where a given entry is in the overlay is required to test this feature. I replaced it with `findEntryIndex` which implements exactly this requirement. It should be harder to misuse because you need to have a reference to the `OverlayEntry` beforehand.